liquibase-core/src/main/java/liquibase/command/core/SnapshotCommandStep.java

```diff
@@ -41,21 +62,135 @@ public class SnapshotCommandStep extends AbstractCliWrapperCommandStep {
                 .setValueObfuscator(ConfigurationValueObfuscator.STANDARD)
                 .build();
         SNAPSHOT_FORMAT_ARG = builder.argument("snapshotFormat", String.class)
-                .description("Output format to use (JSON or YAML").build();
+                .description("Output format to use (JSON, YAML, or TXT)").build();
+        DATABASE_ARG = builder.argument("database", Database.class).hidden().build();
+        SNAPSHOT_CONTROL_ARG = builder.argument("snapshotControl", SnapshotControl.class).hidden().build();
     }
 
+    private Map<String, Object> snapshotMetadata;
+
     @Override
     public String[][] defineCommandNames() {
         return new String[][] { COMMAND_NAME };
     }
 
     @Override
-    protected String[] collectArguments(CommandScope commandScope) throws CommandExecutionException {
-        return collectArguments(commandScope, Arrays.asList(SNAPSHOT_FORMAT_ARG.getName()), null);
+    public void adjustCommandDefinition(CommandDefinition commandDefinition) {
+        commandDefinition.setShortDescription("Capture the current state of the database");
+    }
+
+    public static CatalogAndSchema[] parseSchemas(Database database, String... schemas) {
```

Comment:
Is this a copy of InternalSnapshotCommandStep.parseSchemas? I didn't see it removed from there so it doesn't seem to be a move?

A static utility function like this doesn't really belong on SnapshotCommandStep. It doesn't really belong on InternalSnapshotCommandStep, but since that is "internal" it is less of a stable API we are creating. 

I'd suggest either just leaving it on InternalSnapshotCommandStep until we have a cleaner spot for it, moving this to a utility class (maybe SqlUtil or JdbcUtil?), or making it a protected non-static method for just this class if it's all that's using it.

Reply:
This method can just be private to this class.  It is not used anywhere else.